docs/en/deploy/Running-Alluxio-On-a-HA-Cluster.md

```diff
@@ -0,0 +1,375 @@
+nickname: Cluster with HA
+group: Deploying Alluxio
+priority: 2.5
+---
+
+* Table of Contents
+{:toc}
+
+
+## Overview
+
+An Alluxio service with High availability (or HA) achieved by running multiple Alluxio master processes
+on different nodes in the system. One of these masters is elected the **leading master** which serves all
+workers and clients as the primary point of contact. The other master processes act as **standby
+masters** which maintain the same file system state with the leading master by trailing
+a shared journal. Note that, standby masters do not serve any client or worker requests; however,
+if the leading master fails, one standby master will automatically be chosen to take over and become
+the new leading master. Once the new leading master starts serving, Alluxio clients and workers
+proceed as usual. During the failover to a standby master, clients may experience brief
+delays or transient errors.
+
+The major challenges here to achieve HA are maintaining the shared file system state
+across service restarts and maintain consensus among masters on **leading master**
+after failover. In Alluxio 2.0, there are two different ways to achieve these two goals:
+
+- [Option1](#option1-raft-based-embedded-journal): Use an internal replicated state machine based on RAFT to
+ both store the file system
+journal and select a leading master. This approach is introduced in Alluxio 2.0 and requires no
+dependency on external services.
+- [Option2](#option2-zookeeper-and-shared-journal-storage):
+Leverage an external Zookeeper service for leader election on the leading master
+and a shared storage (e.g., the root UFS) for shared journal.
+See [journal management documentation]({{ '/en/operation/Journal.html' | relativize_url }}) for more information about choosing and configuring Alluxio journal system.
+
+## Prerequisites
+
+* To deploy a Alluxio cluster, first [download](https://alluxio.io/download) the pre-compiled Alluxio
+  binary file, extract the tarball and copy the extracted directory to all nodes (including nodes
+  running masters and workers).
+* Enable SSH login without password from master node to worker nodes. You can add a public SSH key for
+  the host into `~/.ssh/authorized_keys`. See [this tutorial](http://www.linuxproblem.org/art_9.html) for more details.
+* TCP traffic across all nodes is allowed. For basic functionality make sure RPC port (default :19998) is open
+  on all nodes.
+
+## Basic Setup
+### Option1: Raft-based Embedded Journal
+
+The minimal configuration to set up a HA cluster is to give the embedded journal addresses to
+all nodes inside the cluster. On each Alluxio node, create the `conf/alluxio-site.properties` configuration file from the template.
+
+```bash
+cp conf/alluxio-site.properties.template conf/alluxio-site.properties
+```
+
+Add the following properties to the `conf/alluxio-site.properties` file:
+
+```
+alluxio.master.hostname=<MASTER_HOSTNAME> # Only needed on master node
```

Comment:
@calvinjia for worker, do we need to set the quorum for `alluxio.master.hostname` like at the client? or setting `alluxio.master.embedded.journal.addresses` alone is sufficient?

Reply:
`alluxio.master.hostname` should only be necessary on the master when using embedded. Is it needed on the client?

Reply:
I think it is the same on client. But this is confusing in what is required when different options (single master, multi-master with RAFT, multi-master with ZK). We should have a consistent story to set the service address. For now I am fine for the doc. Can you merge it?